packages/pricing/src/services/pricing-module.ts

```diff
@@ -862,6 +862,22 @@ export default class PricingModuleService<
     await this.moneyAmountService_.delete(ids, sharedContext)
   }
 
+  @InjectTransactionManager("baseRepository_")
+  async softDeleteMoneyAmounts(
+    ids: string[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<void> {
+    await this.moneyAmountService_.softDelete(ids, sharedContext)
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async restoreDeletedMoneyAmounts(
+    ids: string[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<void> {
+    await this.moneyAmountService_.restore(ids, sharedContext)
+  }
```

Comment:
todo: can you look at the product module and match the impl, it will facilitate future things ahah :)